# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/extensions/webrequest/WebRequest.jsm
# Commit: 55f13fb4ee3f
# Full Hash: 55f13fb4ee3f06209d00fd91a7a03884bab88e2d
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-12-15 09:27:59
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement csp for Module Workers; r=evilpie,asuth,rpl,ckerschb
#   
#   Depends on D155691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D156102
# ==============================================================================

diff -r 354711cf113a -r 55f13fb4ee3f toolkit/components/extensions/webrequest/WebRequest.jsm
--- a/toolkit/components/extensions/webrequest/WebRequest.jsm	Wed Dec 14 14:55:37 2022 +0000
+++ b/toolkit/components/extensions/webrequest/WebRequest.jsm	Wed Dec 14 14:55:38 2022 +0000
@@ -116,8 +116,10 @@
 
   if (
     isServiceWorkerScript &&
-    channel.loadInfo?.internalContentPolicyType ===
-      Ci.nsIContentPolicy.TYPE_INTERNAL_WORKER_IMPORT_SCRIPTS &&
+    (channel.loadInfo?.internalContentPolicyType ===
+      Ci.nsIContentPolicy.TYPE_INTERNAL_WORKER_IMPORT_SCRIPTS ||
+      channel.loadInfo?.internalContentPolicyType ===
+        Ci.nsIContentPolicy.TYPE_INTERNAL_WORKER_STATIC_MODULE) &&
     !ALLOWED_SERVICEWORKER_SCHEMES.includes(redirectUri?.scheme)
   ) {
     throw new Error(
